diff --git a/board/xilinx/zynq/board.c b/board/xilinx/zynq/board.c
index e7fcf88..8a0685b 100644
--- a/board/xilinx/zynq/board.c
+++ b/board/xilinx/zynq/board.c
@@ -21,8 +21,18 @@
 #include <enclustra_qspi.h>
 #include <enclustra/eeprom-mac.h>
 #include <asm/arch/ps7_init_gpl.h>
-
+#include <asm/io.h>
+#include <asm/gpio.h>
+#include <config.h>
 DECLARE_GLOBAL_DATA_PTR;
+#define CONFIG_SYS_GPIO_0_ADDR                              (0xE000A000)
+/* GPIO Bank 0 */
+#define GPIO_BANK_0_INDX   (0)
+
+/* GPIO Bank 1 */
+#define GPIO_BANK_1_INDX   (1)
+
+#define GPIO_RST_ETH       (50)     // (21)GPIO53
 
 #if (defined(CONFIG_FPGA) && !defined(CONFIG_SPL_BUILD)) || \
     (defined(CONFIG_SPL_FPGA_SUPPORT) && defined(CONFIG_SPL_BUILD))
@@ -213,8 +223,31 @@ static char *zx_get_idcode_name(void)
 }
 #endif
 
+
+
+
+void free_phy_reset(void)
+{
+	puts("Enable phy extern\n");
+	unsigned gpio;
+	gpio = 50;
+	gpio_request(gpio, "cmd_gpio");
+	gpio_direction_output(gpio, 1);
+	udelay(50000);
+	gpio_direction_output(gpio, 0);
+	udelay(50000);
+	gpio_direction_output(gpio, 1);
+	udelay(50000);
+	return;
+}
+
+
+
+
+
 int board_init(void)
 {
+	printf("Swiss timing u-boot 2019\n");
 #if defined(CONFIG_ENV_IS_IN_EEPROM) && !defined(CONFIG_SPL_BUILD)
 	unsigned char eepromsel = CONFIG_SYS_I2C_MUX_EEPROM_SEL;
 #endif
@@ -379,6 +412,11 @@ int board_late_init(void)
 #endif
 #endif
 
+
+	//todo enable PHY 
+	
+		free_phy_reset();
+
 	return 0;
 }
 
